enhancements/cloud-integration/aws/aws-sts-support.md

```diff
@@ -0,0 +1,171 @@
+
+```
+[default]
+aws_secret_access_key = REDACTED
+aws_access_key_id = REDACTED
+```
+
+All pre-existing Secrets will be updated to include this new `credentials` key, and this change will be backported to aid in the upgrade to 4.7, where operators may run before the CCO would have had a chance to add the new key.
+
+The CCO will be modified to set Upgradeable=False for Manual mode users who have not yet updated their Secrets to contain a `credentials` key. This change will need to be backported to 4.6 as well. Alternatively, we could backport a change for these users where CCO automatically generates this new field and includes it, as we have the access key ID and secret access key already in those Secrets, however this may interfere with how manual mode users are deploying/reconciling their Secrets. As such it is probably safest to require them to do it themselves.
+
+#### Operator Changes
+
+All CVO managed operators (~5) that use CredentialsRequests will be updated to use the new `credentials` Secret key. Because this file can also contain a reference to a file with an AWS STS token, operators will then be able to build AWS clients without worrying about what type of authentication they are using.
+
+All CVO managed operators will also now mount in a [Kubernetes Projected Service Account Token](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection) to a consistent file location on disk. (/var/run/secrets/openshift/serviceaccount/token)
+
+```yaml
+      - name: bound-sa-token
+        projected:
+          defaultMode: 420
+          sources:
+          - serviceAccountToken:
+              audience: openshift
+              expirationSeconds: 3600
+              path: token
+```
+
+openshift-install will be modified to accept user provided service account signing keys and configure the Kubernetes APIServer to use them.
+
+For each `CredentialsRequest` the user will need to manually create the relevant Secret and role
+Users will have to manually provision the IAM roles, policies, and Secrets for each `CredentialsRequest`. This process will also require the name of the `ServiceAccount` that the pod consuming the credentials uses, which is typically created by the operator itself. However, we do not presently have the `ServiceAccount` name modeled in the `CredentialsRequest` API, and would like to defer doing this until we have more real world data and experience before making a permanent API commitment. As such, credentials will be documented as a list policy permissions, service account names, and target secrets, which the user will have to iterate:
+
+  1. Create an IAM Role with no policy attached, tagged with `kubernetes.io/cluster/[infraid]=owned` so it can be cleaned up on deprovision. (type = web identity, identity provider created by the admin manually)
```

Comment:
hmm, let's not add the owned tag if the role was not meant for just the cluster. because then the installer will remove it on `destroy cluster`